_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,235 @@
+The new code generator has been very stable for a long time now. The main reason for not
+marking it "non-experimental" earlier was that it still had problems with limited stack space,
+which we were able to fix now.
+
+You can enable the new pipeline using the ``--via-ir`` commandline switch or by setting
+``settings.viaIR`` to ``true`` in Standard JSON input (i.e. at the same level where
+you would have the ``optimizer`` key).
+
+It is not recommended to use the Yul compiler pipeline without the optimizer because the unoptimized output
+is primarily meant to be a straightforward transformation of the high-level Solidity code
+and it has many additional checks compared to the old compiler pipeline.
+This makes it more
+modular and easier to audit while the new Yul-based optimizer is powerful enough to remove redundancies
+from the optimized version.
+
+The performance of the new pipeline is not yet always superior to the old one, but it
+can do much higher-level optimization across functions, so please try it out and give us feedback!
+
+Finally, in some edge cases, code compiled via the new compiler pipeline
+behaves different than code compiled via the old compiler pipeline.
+The most prominent is the initialization order of state variables that depend
+on the result of constructors in other contracts.
+Others are related to weird uses of modifiers, dependence of evaluation order inside
+statements or dirty data. You can find a list of all the changes
+in our [documentation](https://docs.soliditylang.org/en/develop/ir-breaking-changes.html).
+
+### Memory-Safe Assembly / Stack Too Deep
+
+One important feature of the new compiler pipeline is that it can move stack variables to memory
+and thus avoid the "stack too deep" issue. In order to do this safely, the compiler needs to know
+that the memory slots it intends to use are not used by inline assembly.
+
+Inline assembly is considered memory-safe if it only uses memory that has been previously allocated
+either by high-level Solidity code or by reading from the free memory pointer at ``0x40``.
+At the very beginning of the code, the compiler sets the free memory pointer to a position
+after the area reserved for the stack variables that are moved to memory.
+
+Since it is difficult to impossible for the compiler to check that inline assembly only accesses memory in a safe way,
+we added a syntactic mechanism for the developer to assert this:
+
+```
+assembly ("memory-safe") { ... }
+```
+
+Marking an unsafe block as memory-safe can lead to undefined behavior
+due to conflicts between stack variables and memory, the compiler will
+not check that marked blocks are actually memory safe.
+Please see the [documentation](https://docs.soliditylang.org/en/v0.8.13/assembly.html#memory-safety)
+for more information about when an inline assembyl block is memory safe.
+
+By default inline assembly blocks are not considered memory-safe and
```

Comment:
Not strictly true for inline assembly blocks without memory accesses and without references to solidity variables in memory - but not sure we need to mention that, so also fine.

Reply:
I'll add it.